From cb3b05c7af85291bc8fa870f542a6176ee6b126a Mon Sep 17 00:00:00 2001
From: He Chuyue <hechuyue@wxiat.com>
Date: Mon, 16 May 2022 16:34:52 +0800
Subject: [PATCH] sw64: simplify __phys_addr and __virt_addr_valid
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cb3b05c7af85291bc8fa870f542a6176ee6b126a
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

In these methods, the local variable `y` is redundant and can be
removed.

Signed-off-by: He Chuyue <hechuyue@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/mm/physaddr.c | 30 +++++++++++++-----------------
 1 file changed, 13 insertions(+), 17 deletions(-)

diff --git a/arch/sw_64/mm/physaddr.c b/arch/sw_64/mm/physaddr.c
index fbb489ae4db5..26769f0bf7bf 100644
--- a/arch/sw_64/mm/physaddr.c
+++ b/arch/sw_64/mm/physaddr.c
@@ -5,34 +5,30 @@
 
 unsigned long __phys_addr(unsigned long x)
 {
-	unsigned long y = x;
-
-	if (y >= __START_KERNEL_map) {
-		y -= __START_KERNEL_map;
-		VIRTUAL_BUG_ON(y >= KERNEL_IMAGE_SIZE);
+	if (x >= __START_KERNEL_map) {
+		x -= __START_KERNEL_map;
+		VIRTUAL_BUG_ON(x >= KERNEL_IMAGE_SIZE);
 	} else {
-		VIRTUAL_BUG_ON(y < PAGE_OFFSET);
-		y -= PAGE_OFFSET;
-		VIRTUAL_BUG_ON(!phys_addr_valid(y));
+		VIRTUAL_BUG_ON(x < PAGE_OFFSET);
+		x -= PAGE_OFFSET;
+		VIRTUAL_BUG_ON(!phys_addr_valid(x));
 	}
-	return y;
+	return x;
 }
 EXPORT_SYMBOL(__phys_addr);
 
 bool __virt_addr_valid(unsigned long x)
 {
-	unsigned long y = x;
-
-	if (y >= __START_KERNEL_map) {
-		y -= __START_KERNEL_map;
-		if (y >= KERNEL_IMAGE_SIZE)
+	if (x >= __START_KERNEL_map) {
+		x -= __START_KERNEL_map;
+		if (x >= KERNEL_IMAGE_SIZE)
 			return false;
 	} else {
-		if (y < PAGE_OFFSET)
+		if (x < PAGE_OFFSET)
 			return false;
-		y -= PAGE_OFFSET;
+		x -= PAGE_OFFSET;
 	}
 
-	return pfn_valid(y >> PAGE_SHIFT);
+	return pfn_valid(x >> PAGE_SHIFT);
 }
 EXPORT_SYMBOL(__virt_addr_valid);
-- 
2.33.0

